api/v1beta1/network_types.go

```diff
@@ -222,13 +246,21 @@ type SubnetSpec struct {
 	// CidrBlock is the CIDR block to be used when the provider creates a managed VPC.
 	CidrBlock string `json:"cidrBlock,omitempty"`
 
+	// IPv6CidrBlock is the IPv6 CIDR block to be used when the provider creates a managed VPC.
+	// A subnet can have an IPv4 and an IPv6 address.
+	IPv6CidrBlock string `json:"ipv6CidrBlock,omitempty"`
+
 	// AvailabilityZone defines the availability zone to use for this subnet in the cluster's region.
 	AvailabilityZone string `json:"availabilityZone,omitempty"`
 
 	// IsPublic defines the subnet as a public subnet. A subnet is public when it is associated with a route table that has a route to an internet gateway.
 	// +optional
 	IsPublic bool `json:"isPublic"`
 
+	// IsIPv6 defines the subnet as an IPv6 subnet. A subnet is IPv6 when it is associated with a VPC that has IPv6 enabled.
+	// +optional
+	IsIPv6 bool `json:"isIpv6"`
```

Comment:
As this is an optional field
```suggestion
	IsIPv6 bool `json:"isIpv6"`
	IsIPv6 bool `json:"isIpv6,omitempty"`
```